src/core/public/chrome/ui/header/header.tsx

```diff
@@ -147,12 +148,16 @@ export function Header({
   navGroupEnabled,
   setCurrentNavGroup,
   useUpdatedHeader,
+  storage = window.localStorage,
   ...observables
 }: HeaderProps) {
+  const storageKey = `core.leftNav.navGroupEnabled-${!!navGroupEnabled}.isNavOpen`;
   const isVisible = useObservable(observables.isVisible$, false);
   const headerVariant = useObservable(observables.headerVariant$, HeaderVariant.PAGE);
   const isLocked = useObservable(observables.isLocked$, false);
-  const [isNavOpen, setIsNavOpen] = useState(false);
+  const [isNavOpen, setIsNavOpenState] = useState(
+    localStorage.getItem(storageKey) === 'true' ? true : false
```

Comment:
Shall we reuse `isNavDrawerLocked$` in `src/core/public/chrome/chrome_service.tsx`?

Reply:
Good idea, updated.